src/windows/pdb.rs

```diff
@@ -80,45 +81,132 @@ impl PDBSections {
+    fn add_public_symbol(&mut self, symbol: PublicSymbol, address_map: &AddressMap) {
+        self.symbols
+            .add_public_symbol(symbol, &self.pdb_sections, address_map)
+    }
+
+    fn add_procedure_symbol(
+        &mut self,
+        symbol: ProcedureSymbol,
+        info: BlockInfo,
+        lines: &SourceLineCollector,
+    ) -> Result<()> {
+        self.symbols.add_procedure_symbol(lines, symbol, info)
+    }
+
+    fn add_reg_rel(&mut self, symbol: RegisterRelativeSymbol) {
+        // TODO: check that's the correct way to know if we've a parameter here
+        // 22 comes from https://github.com/microsoft/microsoft-pdb/blob/master/include/cvconst.h#L436
+        if self.cpu == CPU::X86 && symbol.register == Register(22 /* EBP */) && symbol.offset > 0 {
+            self.symbols.add_ebp(symbol);
+        }
+    }
+
+    fn close_procedure(&mut self) {
+        self.symbols.close_procedure();
+    }
+}
+
+pub(crate) struct PDBInfo {
+    symbols: PDBSymbols,
+    files: Vec<String>,
     cpu: CPU,
     debug_id: String,
-    pe_name: String,
     pdb_name: String,
-    source_files: SourceFiles<'s>,
-    pdb_sections: PDBSections,
-    rva_symbols: RvaSymbols,
-    address_map: AddressMap<'s>,
+    pe_name: String,
+    code_id: Option<String>,
+    stack: Option<String>,
 }
 
-impl PDBInfo<'_> {
-    fn get_cpu(dbi: &DebugInformation) -> CPU {
-        if let Ok(mt) = dbi.machine_type() {
-            match mt {
-                // Currently breakpad code only uses these machine types
-                // but we've more possibilities:
-                // https://docs.rs/pdb/0.5.0/pdb/enum.MachineType.html
-                MachineType::X86 => CPU::X86,
-                MachineType::Amd64 | MachineType::Ia64 => CPU::X86_64,
-                _ => CPU::Unknown,
-            }
-        } else {
-            CPU::Unknown
+impl Display for PDBInfo {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        writeln!(
+            f,
+            "MODULE windows {} {} {}",
+            self.cpu, self.debug_id, self.pdb_name
+        )?;
+
+        if let Some(code_id) = self.code_id.as_ref() {
+            writeln!(f, "INFO CODE_ID {} {}", code_id, self.pe_name)?;
+        }
+
+        for (n, file_name) in self.files.iter().enumerate() {
+            writeln!(f, "FILE {} {}", n, file_name)?;
+        }
+
+        for (_, sym) in self.symbols.iter() {
+            write!(f, "{}", sym)?;
+        }
+
+        if let Some(stack) = self.stack.as_ref() {
+            write!(f, "{}", stack)?;
+        }
+
+        Ok(())
+    }
+}
+
+fn get_cpu(dbi: &DebugInformation) -> CPU {
+    if let Ok(mt) = dbi.machine_type() {
+        match mt {
+            // Currently breakpad code only uses these machine types
+            // but we've more possibilities:
+            // https://docs.rs/pdb/0.5.0/pdb/enum.MachineType.html
+            MachineType::X86 => CPU::X86,
+            MachineType::Amd64 | MachineType::Ia64 => CPU::X86_64,
```

Comment:
Open an issue to add the `Arm64` CPU type, we'll need it.

Reply:
Done: https://github.com/mozilla/dump_syms/issues/21